src/metrics.jl

```diff
@@ -12,13 +12,21 @@ abstract type UnionSemiMetric <: SemiMetric end
 
 abstract type UnionMetric <: Metric end
 
+# a minkowski metric is a metric that is defined by the formula:
```

Comment:
Make this a docstring (with appropriate formatting)? It would be useful to explain how this differs from `Minkowski` as it could confuse users. Could also be interesting to say what values of `p` subtypes correspond to.

Also:
```suggestion
# a Minkowski metric is a metric that is defined by the formula:
```

Reply:
Done!

Reply:
@nalimilan, do you think this new type should be exported?

Reply:
Good question. In theory I would say yes, but the fact that `MinkowskiMetric <: UnionMetric` is problematic to me, as the latter is internal and its interface isn't documented (see https://github.com/JuliaStats/Distances.jl/issues/151#issuecomment-612460855 for how it apparently works).

Actually this raises a more fundamental point: in the current state of the PR, if people want to implement a `MinkowskiMetric` subtype, they have to implement the `UnionMetric` interface, or at least define enough methods so that none of the fallback methods defined for that type are used. This isn't logical. It would seem better to introduce `UnionMinkowskiMetric <: MinkowskiMetric <: PreMetric` and redefine `UnionMetrics` to include that (see https://github.com/JuliaStats/Distances.jl/pull/163 for context about `UnionMetric`). What do you think @dkarrasch?

Reply:
@dkarrasch, the PR was like this at the beginning:

```julia
# /src/generic.jl
abstract type MinkowskiMetric <: Metric end

# /src/metrics.jl
abstract type UnionMinkowskiMetric <: MinkowskiMetric end
```

But after some discussions this was changed to the current code.

Reply:
Ah sorry I had missed that discussion. This is definitely tricky. @dkarrasch You said:

> Couldn't you just subtype `MinkowskiMetric <: UnionMetric`? Thinking about it more, I think you actually should do that, because otherwise these types fall out of the `UnionMetric` type branch, which potentially other users rely on.

But why do you think it would be a problem that these types would not be `UnionMetric`? This is internal anyway. It seems to me that according to your reasoning, we could drop all `Union*` types, e.g. replacing `UnionMetric` with just `Metric`.

Reply:
After second thought, I think @nalimilan's proposal (which seems like the original state of the PR) is better. I confused `UnionMetric` with `UnionMetrics`, and what I was having in mind as the `UnionMetric` mechanism/infrastructure (the `eval_*` stuff) is actually a  `UnionMetrics` mechanism. Sorry for spreading confusion, and thanks @nalimilan for your care- and thoughtful review!

Reply:
@eliascarv Can you bump the patch version right in this PR? Then after we merge, we can release right away.

Reply:
@nalimilan, so we can export the type `MinkowskiMetric` (not `UnionMinkowskiMetric`), right?

Reply:
Yes I think so.